Entities/Items/Projectiles/Arrow.as

```diff
@@ -582,6 +598,24 @@ void ArrowHitMap(CBlob@ this, Vec2f worldPoint, Vec2f velocity, f32 damage, u8 c
 	Vec2f lock = worldPoint - norm;
 	this.set_Vec2f("lock", lock);
 
+	// saving information on what was hit to determine when the arrow should collapse
+	
+	if (hitBlob is null) // map (stone, wood, dirt)
+	{
+		this.set_bool("hitBlobExists", false);
+		this.set_Vec2f("hitWorldPoint", worldPoint);
+		
+		this.Sync("hitWorldPoint", true);
+	}
+	else // hitBlob (door, platform, etc.)
+	{
+		this.set_bool("hitBlobExists", true);
+		this.set_u32("hitBlob", hitBlob.getNetworkID());
+		
+		this.Sync("hitBlob", true);
+	}
+	
+	this.Sync("hitBlobExists", true);
```

Comment:
Since we're only setting vars to kill this blob (server-side) eventually, put a ``isServer()`` check for lines 601 to 618. And since we are only doing server-side anyways, you don't need the Sync() instances, so remove all these syncs you added since they aren't used for anything good.